### ISSUE TITLE
S3928 "ArgumentException constructor arguments have been inverted" triggered with named parameters

### ISSUE THREAD

**Issue by Duranom** [SonarSource/sonar-dotnet#5226]
### Description

Please provide a succinct description of your issue. Also please provide the rule ID (e.g. S1234)
Concerning: S3928
When using ArgumentException with named parameters in different order the S3928 still triggers

### Repro steps
Sample of case that triggers it:
```
using System;

namespace Sample;
public class Program
{
    public static void Main(string?[] args)
    {
        // The following line triggers S3928
        throw new ArgumentException(paramName: nameof(args), message: "Sample message");
    }
}
```

### Expected behavior

Because named parameters are used the rule S3928 is not expected to be triggered as the developer is aware and uses it to clarify.

### Actual behavior

Rule S3928 is triggered by SonarQube and SonarLint
The rule cannot be disabled as ArgumentExceptions that were made without named parameters still need to be detected

### Known workarounds

Switching the named parameters in the actual place of the method fixes it, but for hardcoded messages where no culture or other ways are needed or deep in depth resources references it because hard to read while only wanted to quickly see the parameter name.

### Related information

Related information
Sonarqube: Version 8.9.6 (build 50800) and higher
Visual Studio Version: VS 2022
SonarLint: SonarLint for Visualstudio 2022 v5.3.0.41207
